app/src/main/java/net/bible/service/db/readingplan/ReadingPlanDBAdapter.kt

```diff
@@ -0,0 +1,767 @@
+/*
+ * Copyright (c) 2019 Martin Denham, Tuomas Airaksinen and the And Bible contributors.
+ *
+ * This file is part of And Bible (http://github.com/AndBible/and-bible).
+ *
+ * And Bible is free software: you can redistribute it and/or modify it under the
+ * terms of the GNU General Public License as published by the Free Software Foundation,
+ * either version 3 of the License, or (at your option) any later version.
+ *
+ * And Bible is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with And Bible.
+ * If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+package net.bible.service.db.readingplan
+
+import android.content.ContentValues
+import android.provider.BaseColumns
+import android.util.Log
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.SerializationException
+import kotlinx.serialization.json.JSON
+import net.bible.android.BibleApplication
+import net.bible.android.activity.R
+import net.bible.android.view.activity.readingplan.DailyReading
+import net.bible.service.common.CommonUtils
+import net.bible.service.db.CommonDatabaseHelper
+import net.bible.service.db.readingplan.ReadingPlanDatabaseDefinition.DB_FALSE_VALUE
+import net.bible.service.db.readingplan.ReadingPlanDatabaseDefinition.DB_TRUE_VALUE
+import net.bible.service.db.readingplan.ReadingPlanDatabaseDefinition.ReadingPlanDays
+import net.bible.service.db.readingplan.ReadingPlanDatabaseDefinition.ReadingPlanMeta
+import net.bible.service.readingplan.PassageReader
+import org.apache.commons.lang3.StringUtils
+import org.crosswire.jsword.passage.Key
+import org.crosswire.jsword.versification.Versification
+import org.crosswire.jsword.versification.system.SystemNRSVA
+import org.crosswire.jsword.versification.system.Versifications
+import java.text.SimpleDateFormat
+import java.util.Calendar
+import java.util.Date
+import kotlin.collections.ArrayList
+
+/**
+ * @author Timmy Braun [tim.bze at gmail dot com] (2/14/2019)
+ */
+class ReadingPlanDBAdapter {
+    companion object {
+        const val TAG = "ReadingPlanDBAdapter"
+        const val INCLUSIVE_VERSIFICATION = SystemNRSVA.V11N_NAME
+        val app = BibleApplication.application
+        val dbHelper: CommonDatabaseHelper = CommonDatabaseHelper.getInstance()
+
+        private val dbWritable = dbHelper.writableDatabase
```

Comment:
dbHelper has a writable and readable DB object. One is Read/Write, other is read-only. I think in this place it only needs to be set at initialization. Maybe I'm wrong.

Reply:
Okay, I see. getReadableDatabase seems to give writable database as well if possible, it just does not fail like getWritableDatabase would if it is not possible to open it in read-write mode (so I understood from the documentation of these functions). So to me it looks like only one would be enough. Though I don't know best conventions here.

Reply:
Ok I didn't know that readableDatabase also gets read/write. I will just use that.

Is it better to use it as a property? I'm just wondering cause the documentation says it could take long to load (even though for such small db file it shouldn't be too long). I'm not sure if it's good to constantly be loading it then.

Reply:
I think constant indeed is correct way after all.